pkg/deploy/chart_manager.go

```diff
@@ -6,26 +6,28 @@ import (
 	"fmt"
 	"net"
 	"strconv"
-	"time"
+
+	moduletypes "github.com/kyma-project/module-manager/operator/pkg/types"
+
+	corev1 "k8s.io/api/core/v1"
 
 	"github.com/kyma-project/lifecycle-manager/api/v1alpha1"
 	"github.com/kyma-project/lifecycle-manager/pkg/remote"
 	modulelib "github.com/kyma-project/module-manager/operator/pkg/manifest"
-	corev1 "k8s.io/api/core/v1"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	k8syaml "sigs.k8s.io/yaml"
 )
 
+const caCertificateSecretKey = "ca.crt"
+
 var ErrExpectedExactlyOneSKRConfig = errors.New("expected exactly one SKR config")
 
 type SKRWebhookChartManager interface {
-	InstallWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
-		remoteClientCache *remote.ClientCache, kcpClient client.Client) (bool, error)
-	RemoveWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
-		syncCtx *remote.KymaSynchronizationContext) error
+	InstallWebhookChart(ctx context.Context, syncContext *remote.KymaSynchronizationContext) (bool, error)
+	RemoveWebhookChart(ctx context.Context, syncContext *remote.KymaSynchronizationContext) error
 }
 
 type DisabledSKRWebhookChartManager struct{}
```

Comment:
Imo this implementation should change to not use a `Disabled` struct. It should not be created in the first place. This kind of design reminds me a lot of Java interface implementations...